### DISCUSSION TITLE
The convolution implemented in CUTLASS on SM90 is through the im2col, Instead of implicit precom gemm ?

### DISCUSSION THREAD

**Discussion by luliyucoordinate** [NVIDIA/cutlass#1423] [Q&A]
Currently, the convolution implemented in CUTLASS on SM90 is through the im2col method, while for SM architectures less than SM90, the implicit precompute GEMM approach is used. Why? And, how can I implement the implicit precompute GEMM CONV using cute ?

**Comment by hwu36** [comment]
sm90 still use implicit gemm, but it uses tma to do complicated address computation and boundary check. 

https://github.com/NVIDIA/cutlass/tree/main/examples/59_ampere_gather_scatter_conv shows how to use cute to do conv.
